### PR TITLE
Add support for Source Maps

### PR DESCRIPTION
**Pull request by horka**
Added source map generation functionality to the task. Also modified the tests to test the source map generation, and added info about this option to the README.md file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +42,40 @@ Report minification result or both minification and gzip results.
 This is useful to see exactly how well clean-css is performing but using `'gzip'` will make the task take 5-10x longer to complete. [Example output](https://github.com/sindresorhus/maxmin#readme).
 
 
+#### sourceMap
```
Comment (sindresorhus) [inline README.md]: Needs to be added to `docs/options.md` too as the readme is generated.

File: README.md [hunk 2]
```diff
@@ -29,7 +29,8 @@ _Run this task with the `grunt cssmin` command._
 
 ### Options
 
-Options are passed to [clean-css](https://github.com/jakubpawlowicz/clean-css#how-to-use-clean-css-programmatically). In addition this task defines an extra option:
+Options are passed to [clean-css](https://github.com/jakubpawlowicz/clean-css#how-to-use-clean-css-programmatically).
+This task defines two options by default: report and sourceMap.
```
Comment (sindresorhus) [inline README.md]: Previous sentence was better.

File: README.md [hunk 3]
```diff
@@ -41,6 +42,40 @@ Report minification result or both minification and gzip results.
...
+      dest: 'release/css',
+      ext: '.min.css'
+    }],
+    options: {
+      shorthandCompacting: false,
+      roundingPrecision: -1
+    }
+  }
+}
+```
```
Comment (sindresorhus) [inline README.md]: No point in having an example for this.

File: tasks/cssmin.js [hunk 4]
```diff
@@ -2,7 +2,7 @@
  * grunt-contrib-cssmin
  * http://gruntjs.com/
  *
- * Copyright (c) 2015 Tim Branyen, contributors
+ * Copyright (c) 2014 Tim Branyen, contributors
```
Comment (sindresorhus) [inline tasks/cssmin.js]: ?

File: Gruntfile.js [hunk 5]
```diff
@@ -23,6 +23,9 @@ module.exports = function(grunt) {
       test: ['tmp']
     },
     cssmin: {
+    options: {
+      sourceMap: true
+    },
```
Comment (sindresorhus) [inline Gruntfile.js]: wrong indentation

File: tasks/cssmin.js [hunk 6]
```diff
@@ -13,46 +13,54 @@ var chalk = require('chalk');
...
+    var getAvailableFiles = function (filesArray) {
+        return filesArray.filter(function(filepath) {
+            if (!grunt.file.exists(filepath)) {
+                grunt.log.warn('Source file ' + chalk.cyan(filepath) + ' not found.');
+                return false;
+            } else {
+                return true;
+            }
+        });
+    };
```
Comment (sindresorhus) [inline tasks/cssmin.js]: Don't change the indentation.

**Comment by horka** [comment]
Thanks for the review notes, made the changes. Don't know why changed the date, sorry ;)

File: tasks/cssmin.js [hunk 7]
```diff
@@ -13,46 +13,53 @@ var chalk = require('chalk');
...
+      if (options.sourceMap) {
+        var compiledFileName = file.dest.split('/').pop();
+        var sourceMapString = compiled.sourceMap.toString();
+        compiledCssString += '\n' + '/*# sourceMappingURL='+compiledFileName+'.map */';
+        grunt.file.write(file.dest+'.map', sourceMapString);
+      }
 
-      grunt.file.write(file.dest, min);
-
-      grunt.verbose.writeln('File ' + chalk.cyan(file.dest) + ' created: ' + maxmin(max, min, options.report === 'gzip'));
```
Comment (sindresorhus) [inline tasks/cssmin.js]: Don't remove the maxmin reporting.

File: tasks/cssmin.js [hunk 8]
```diff
@@ -13,46 +13,53 @@ var chalk = require('chalk');
...
-
-      var max = '';
-      var min = valid.map(function(file) {
-        var src = grunt.file.read(file);
-        max += src;
-        options.relativeTo = path.dirname(file);
-        return minify(src, options);
-      }).join('');
+      var availableFiles = getAvailableFiles(file.src);
+      var cleanCssOptions = { sourceMap: true, target: file.dest };
```
Comment (sindresorhus) [inline tasks/cssmin.js]: The `sourceMap` option should adhere to the user specified `sourceMap` option.

**Comment by horka** [comment]
Thanks for the fast review, fixed both.

**Pull request closed without merging**